### ISSUE TITLE
[java] LocalVariableCouldBeFinal shouldn't report unused variables

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4511]
**Affects PMD Version:** 6.x

**Rule:** [LocalVariableCouldBeFinal](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#localvariablecouldbefinal)

**Description:**

A local variable, that is just defined but not used, is a different problem: [UnusedLocalVariable](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable). Make the variable `final` doesn't fix anything.

Found via #3123. It's already fixed with PMD 7.0.0-rc1.
Was fixed via #3113.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public void test1() {
        int a = 0; // not final, but also not used
    }
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.
